src/chromo/kinematics.py

```diff
@@ -230,11 +230,41 @@ def __init__(
 
         _gamma_cm = (elab + m2) / ecm
         _betagamma_cm = plab / ecm
+        self.m1 = m1
+        self.m2 = m2
 
         super().__init__(
             frame, part1, part2, ecm, plab, elab, ekin, beams, _gamma_cm, _betagamma_cm
         )
 
+        self._set_beam_data()
+
+    @staticmethod
+    def _get_beam_data(kin):
+        return BeamData(
+            ref_frame=kin.frame,
+            pid=np.array([int(kin.p1), int(kin.p2)]),
+            status=np.array([4, 4]),
+            charge=np.array([kin.p1.charge, kin.p2.charge]),
+            px=np.zeros((2,), dtype=np.float64),
+            py=np.zeros((2,), dtype=np.float64),
+            pz=np.array([kin.beams[0][2], kin.beams[1][2]]),
+            en=np.array([kin.beams[0][3], kin.beams[1][3]]),
+            m=np.array([kin.m1, kin.m2]),
+            vx=np.zeros((2,), dtype=np.float64),
+            vy=np.zeros((2,), dtype=np.float64),
+            vz=np.zeros((2,), dtype=np.float64),
+            vt=np.zeros((2,), dtype=np.float64),
+            mothers=np.array([[-1, -1], [-1, -1]], dtype=np.int32),
```

Comment:
Beam particles never have mothers and you cannot set the daugthers before you know the actual event. Neither attribute should be part of the dataclass.

Reply:
Anyway, there should be some value assigned in the end. It is convenient to have beam record that has the same structure as event.